OpenUpSA/wazimap-ng-ui
# navigating to another geography after filtering choropleth

## Description
After displaying choropleth for an indicator with a default filter(or a non-aggregatable group), if the user navigates to another geography we do not reset `originalChildData`. This causes us to lose the real data. After that, if the user changes the filter, we show NaN values

## Related Issue
https://wazimap.atlassian.net/browse/WNCM-393

## How is it tested automatically?
added test steps into `data_mapper.feature`

## How should a reviewer test it locally
* Select an indicator with a default filter / non-aggregatable group e.g Economic Opportunities > NEET Status > NEET
* Navigate to a child geography
* Filter on a different value
* Confirm that the map still has color and the choropleth shows the correct values

## Screenshots
before the fix :
![image](https://user-images.githubusercontent.com/53019884/169920765-ffee2029-5dcb-42ac-bf85-bd330c073363.png)

after the fix : 
![image](https://user-images.githubusercontent.com/53019884/169920812-86aaf80b-1fd2-41fb-8e22-a056e9c8ad73.png)



## Changelog

### Added

### Updated
* `src/js/versions/version_controller.js` to make sure `versions.indicators.ready` event is not triggered more than once
* `src/js/setup/choropleth.js` to reset `originalChildData` variable when navigating to a new geography

### Removed


## Checklist


### Pull Request


### Code Quality


### Testing
